docs/enhancements/0001-worker-pods.md

```diff
@@ -0,0 +1,232 @@
+- the operator;
+- the worker Pods.
+
+```
+               ┌───────┐
+               │       │
+               │ Nodes ◄──┐
+               │       │  │
+               └───────┘  │ watches
+                          │
+             ┌─────────┐  │
+             │         │  │
+             │ Modules │  │     ┌───────────────────────┐
+             │         │  │     │                       │
+             └─────▲───┘  │     │ ┌───────────────────┐ │
+                   │      │     │ │                   │ │  creates & owns
+        watches &  └──────┴─────┼─┤ Module controller ├─┼──────────────────┐
+        updates status          │ │                   │ │                  │
+                                │ └───────────────────┘ │           ┌──────▼──────┐
+                                │                       │           │             │
+                                │                       │           │ NodeModules │
+┌────────────┐                  │ ┌─────────────┐       │           │             │
+│            │                  │ │             │       │           └──────▲──────┘
+│ KMM worker ◄──────────────────┼─┤ NodeModules │       │                  │
+│    Pod     │  creates & owns  │ │ controller  ├───────┼──────────────────┘
+│            │                  │ │             │       │   watches &
+└──────┬─────┘                  │ └─────────────┘       │   updates status
+       │                        │                       │
+       │                        │          KMM operator │
+       │                        └───────────────────────┘
+       │
+       │
+       │                        ┌──────────────────────┐
+       │                        │                      │
+       └────────────────────────► registry.example.org │
+                pulls from      │                      │
+                                └──────────────────────┘
+```
+
+#### The operator
+
+The operator runs two main controllers: the `Module` controller and the `NodeModules` controller.
+
+The `Module` controller watches both `Module` and `Node` resources to determine which kmods should be loaded on which
+`Node`.
+It creates one `NodeModules` resource per `Node`.
+`NodeModules` resources represent the desired state of all `Modules` targting the node; they contain all the required
+data to pull kmod images and load the modules properly.  
+The `Module` controller watches updates to the `status` of `NodeModules` objects and aggregates that data into the
+`Module` `status`.
+
+The `NodeModules` controller watches `NodeModules` resources and determines when a worker Pod needs to be started on a
+node by looking at the difference between the desired state (spec) and the current one (status).
+When the Pod has completed, the controller reads its outcome and updates the `NodeModules` status accordingly.
+
+When updating the `NodeModules` status, the controller includes a timestamp corresponding to the last time a worker Pod
```

Comment:
I think that we need to add (somewhere) the correlation between NodeModules and worker Pods, meaning: we will be creating one Pod per Node per Module, just to make it clear